---
float32[42] data